src/Bridges/bridge_optimizer.jl

```diff
@@ -1241,6 +1244,42 @@ function MOI.supports(
     )
 end
 
+function _set_substituted(
+    b::AbstractBridgeOptimizer,
+    attr::MOI.ConstraintSet,
+    ci::MOI.ConstraintIndex{F,S},
+    value,
+) where {F,S}
+    if !(value isa S)
+        throw(ArgumentError("Invalid type when setting ConstraintSet."))
+    elseif is_bridged(b, ci)
+        MOI.throw_if_not_valid(b, ci)
+        call_in_context(MOI.set, b, ci, attr, value)
+    else
+        MOI.set(b.model, attr, ci, value)
+    end
+    return
+end
+
+function _set_substituted(
+    b::AbstractBridgeOptimizer,
+    attr::MOI.ConstraintFunction,
+    ci::MOI.ConstraintIndex{F,S},
+    value,
+) where {F,S}
+    if F == MOI.VariableIndex
+        throw(MOI.SettingVariableIndexNotAllowed())
```

Comment:
Why not adding a method
```julia
function MOI.set(
     b::AbstractBridgeOptimizer,
    attr::MOI.ConstraintFunction,
    ci::MOI.ConstraintIndex{MOI.SingleVariable},
    func::MOI.SingleVariable,
)
```
I prefer keeping only one `_set_substituted` and doing the dispatch at the `MOI.set` level.